File: exercism/python/grains/grains.py
def square(number):

    if 1 <= number <= 64:
        count = 0
        for a in range(number):
            c = a + 1
            if c == 1:
                count = 1
            elif c == 2:
                count = 2
            else:
                count = count * 2
        return count
    else:
        # when the square value is not in the acceptable range        
        raise ValueError("square must be between 1 and 64")


def total():
    count = 0
    for a in range(64):
        count += square(a+1)
    return count

File: exercism/typescript/grains/grains.ts
export const square = (number: bigint): bigint => {
  if (BigInt(1) <= number && number <= BigInt(64)) {
    let count: bigint = BigInt(0);
    for (let a: bigint = BigInt(0); a < number; a++) {
      const c: bigint = a + BigInt(1);
      if (c === BigInt(1)) {
        count = BigInt(1);
      } else if (c === BigInt(2)) {
        count = BigInt(2);
      } else {
        count = count * BigInt(2);
      }
    }
    return count;
  } else {
    // when the square value is not in the acceptable range
    throw new Error("square must be between 1 and 64");
  }
}

export const total = (): bigint => {
   let count: bigint = BigInt(0);
  for (let a: bigint = BigInt(0); a < BigInt(64); a++) {
    count += square(a + BigInt(1));
  }
  return count;
}


